packages/quick_actions/quick_actions_android/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java

```diff
@@ -97,26 +115,20 @@ public boolean onNewIntent(@NonNull Intent intent) {
       return false;
     }
     // Notify the Dart side if the launch intent has the intent extra relevant to quick actions.
-    if (intent.hasExtra(MethodCallHandlerImpl.EXTRA_ACTION) && channel != null) {
-      Context context = activity.getApplicationContext();
+    if (intent.hasExtra(QuickActions.EXTRA_ACTION)) {
+      Context context = this.quickActions.getActivity().getApplicationContext();
       ShortcutManager shortcutManager =
           (ShortcutManager) context.getSystemService(Context.SHORTCUT_SERVICE);
-      String shortcutId = intent.getStringExtra(MethodCallHandlerImpl.EXTRA_ACTION);
-      channel.invokeMethod("launch", shortcutId);
+      String shortcutId = intent.getStringExtra(QuickActions.EXTRA_ACTION);
+      quickActionsFlutterApi.launchAction(
+          shortcutId,
+          new AndroidQuickActionsFlutterApi.Reply<Void>() {
```

Comment:
Since this is a single method, can't we replace the whole thing with a lambda? (If so, AS will offer to do it for you automagically via a tip.)

Reply:
it isn't suggesting anything, so 🤷